fvcore/common/checkpoint.py

```diff
@@ -290,6 +298,15 @@ def step(self, iteration: int, **kwargs: Any):
         if iteration >= self.max_iter - 1:
             self.checkpointer.save("model_final", **additional_state)
 
+        if self.n_keep is not None:
```

Comment:
Could you move the whole logic into 
```
if (iteration + 1) % self.period == 0:
```
?

The deletion logic needs to list a directory and therefore potentially slow, based on the underlying storage system. It is sufficient to only run it after saving a new checkpoint, and it's not necessary to worry about an extra "model_final".

Reply:
Done.